packages/flutter/lib/src/material/search.dart

```diff
@@ -89,6 +89,23 @@ Future<T> showSearch<T>({
 /// for another [showSearch] call.
 abstract class SearchDelegate<T> {
 
+  /// The type of action button to use for the keyboard.
```

Comment:
We could keep the fields `final` by using a constructor instead like this:

```dart
  const SearchDelegate({
    this.keyboardType,
    this.hintText,
    this.textInputAction = TextInputAction.search,
  });
```

Then below, make all of the fields `final` and remove the default assignment from `textInputAction`.